include/deal.II/base/memory_consumption.h

```diff
@@ -631,6 +641,16 @@ namespace MemoryConsumption
   template <typename T>
   inline
   std::size_t
+  memory_consumption (const std_cxx11::unique_ptr<T> &)
+  {
+    return sizeof(std_cxx11::unique_ptr<T>);
```

Comment:
Here I'm wondering whether you shouldn't also include the memory consumed of T. WIth a unique_ptr, it is clear that the memory is where the pointer is. (shared_ptr is a different story.) But I'll have to check below how you use it.